internals/__init__.py

```diff
@@ -61,11 +68,16 @@ class QuarantineStatus(object):
     def __init__(self, energy: int, fulfillment: int, action_history: List[Action]):
         self.energy = energy
         self.fulfillment = fulfillment
+        self.time_gen = advance_time()
+        self.current_time = next(self.time_gen)
         self.action_history = action_history
 
     # When applying an action, get the Action object from the global ACTIONS
     # dict: `state.apply_action(ACTIONS["drink_beer"])`
-    def apply_action(self, action_name: str):
+    def apply_action(self, action_name: str) -> bool:
         action: Action = ACTIONS[action_name]
         self.energy += action.delta_energy
         self.fulfillment += action.delta_fulfillment
+        self.current_time = next(self.time_gen)
+        self.action_history.append(action)
```

Comment:
this stuff will have to get reconciled with my PR but other than my one req lgtm